### PR TITLE
Introduce kyma-developer role

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Aggregate kyma-view to view role
- Create kyma-edit role, and aggregate it to edit
- Create kyma-developer role and binding, which use the extended edit role
- Create static dveloper-user in dex for testing

**Related issue(s)**

Implements #2557

### CODE REVIEW

File: resources/dex/templates/dex-users-secret.yaml [hunk 1]
```diff
@@ -50,4 +51,20 @@ data:
...
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": "pre-install"
+  labels:
+    "dex-user-config": "true"
+data:
+  # developer@kyma.cx
+  email: ZGV2ZWxvcGVyQGt5bWEuY3g=
+  # developer-user
+  username: ZGV2ZWxvcGVyLXVzZXI=
```
Comment (pbochynski) [inline resources/dex/templates/dex-users-secret.yaml]: I see test user, I don't see a test...

File: docs/security/docs/08-02-manage-static-users.md [hunk 2]
```diff
@@ -40,7 +40,12 @@ If you want to add a new static user after Dex is installed, restart the Dex Pod
 ## Bind a user to a Role or a ClusterRole
 
 A newly created static user has no access to any resources of the cluster as there is no Role or ClusterRole bound to it.  
-By default, Kyma comes with two ClusterRoles: **kyma-admin**, which allows to get full admin access to the entire cluster, and **kyma-view**, which allows to view and list all of the resources of the cluster.
+By default, Kyma comes with the following ClusterRoles:
+ 
+- **kyma-admin**: allows to get full admin access to the entire cluster
+- **kyma-edit**: allows full access to kyma managed resources
```
Comment (tomekpapiernik) [inline docs/security/docs/08-02-manage-static-users.md]:
```suggestion
- **kyma-edit**: allows full access to all Kyma-managed resources
```

File: docs/security/docs/08-02-manage-static-users.md [hunk 3]
```diff
@@ -40,7 +40,12 @@ If you want to add a new static user after Dex is installed, restart the Dex Pod
 ## Bind a user to a Role or a ClusterRole
 
 A newly created static user has no access to any resources of the cluster as there is no Role or ClusterRole bound to it.  
-By default, Kyma comes with two ClusterRoles: **kyma-admin**, which allows to get full admin access to the entire cluster, and **kyma-view**, which allows to view and list all of the resources of the cluster.
+By default, Kyma comes with the following ClusterRoles:
+ 
+- **kyma-admin**: allows to get full admin access to the entire cluster
```
Comment (tomekpapiernik) [inline docs/security/docs/08-02-manage-static-users.md]:
```suggestion
- **kyma-admin**: allows full admin access to the entire cluster
```

File: docs/security/docs/08-02-manage-static-users.md [hunk 4]
```diff
@@ -40,7 +40,12 @@ If you want to add a new static user after Dex is installed, restart the Dex Pod
...
+- **kyma-edit**: allows full access to kyma managed resources
+- **kyma-developer**: **kyma-edit** + full access to base kubernetes resources
```
Comment (tomekpapiernik) [inline docs/security/docs/08-02-manage-static-users.md]:
```suggestion
- **kyma-developer**:  allows full access to Kyma-managed resources and basic Kubernetes resources
```

Comment (tomekpapiernik) [reply docs/security/docs/08-02-manage-static-users.md]: I think you should list these Kubernetes resources here. 
`(...)and basic Kubernetes resources, such as (...)`

File: docs/security/docs/08-02-manage-static-users.md [hunk 5]
```diff
@@ -40,7 +40,12 @@ If you want to add a new static user after Dex is installed, restart the Dex Pod
...
+- **kyma-view**: allows to view and list all of the resources of the cluster.
```
Comment (tomekpapiernik) [inline docs/security/docs/08-02-manage-static-users.md]:
```suggestion
- **kyma-view**: allows to view and list all of the resources of the cluster
```

File: tests/cluster-users-test/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+MAINTAINER jakub.blaszczyk@sap.com
+
+WORKDIR /root
+
+RUN apt-get update &&\
+	apt-get install -y curl &&\
+	curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl &&\
+	chmod +x kubectl &&\
+	mv kubectl /usr/local/bin/kubectl
+
```
Comment (kubadz) [inline tests/cluster-users-test/Dockerfile]: missing label
```suggestion
LABEL source=git@github.com:kyma-project/kyma.git
```

File: resources/dex/templates/dex-users-secret.yaml (hunk 1 above)
Comment (Demonsthere) [reply resources/dex/templates/dex-users-secret.yaml]: Now you should see some :)

File: tests/cluster-users-test/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,13 @@
+FROM debian:stretch-slim
+MAINTAINER jakub.blaszczyk@sap.com
```
[the remark this answers is not in the document]
Comment (Demonsthere) [reply tests/cluster-users-test/Dockerfile]: Ok good idea

File: resources/core/charts/cluster-users/templates/tests/test-cluster-users.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{{ if .Values.global.isLocalEnv }}
```
[the remark this answers is not in the document]
Comment (Demonsthere) [reply resources/core/charts/cluster-users/templates/tests/test-cluster-users.yaml]: Setting this resulted in running the tests on kyma-integration prowjob on minikube, which ended in a success. The tests were skipped in kyma-gke-integration. Is this enough?

File: resources/core/charts/cluster-users/templates/rbac-roles.yaml [hunk 9]
```diff
@@ -124,3 +125,78 @@ subjects:
...
+    "helm.sh/hook-weight": "0"
+aggregationRule:
+  clusterRoleSelectors:
+  - matchLabels:
+      rbac.authorization.k8s.io/aggregate-to-edit: "true"
+  - matchLabels:
+      rbac.authorization.kyma-project.io/aggregate-to-kyma-edit: "true"
+rules: []
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (pbochynski) [inline resources/core/charts/cluster-users/templates/rbac-roles.yaml]: Kyma developer role should be granted only in namespaces. The developer should not be able to create deployment in kyma-system. Please do not create this binding. In the test, you should grant this role in one namespace and test access in that namespace (you can use default).

Comment (Demonsthere) [reply resources/core/charts/cluster-users/templates/rbac-roles.yaml]: Moved Binding into test, added namespace as an env to the test

File: tests/cluster-users-test/sar-test.sh [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+	if [[ "${TEST}" == "${EXPECTED}" ]]; then
+		echo "----> PASSED"
+		return 0
+	fi
+	echo "----> |FAIL| Expected result: ${EXPECTED} vs Actual result: ${TEST}"
+	return 1
+}
+
+function runTests() {
+	echo "--> developer@kyma.cx should be able to get Deployments"
```
Comment (pbochynski) [inline tests/cluster-users-test/sar-test.sh]: Can you add tests showing that the developer can access namespaced resources (pods, deployments) in the namespace he has rolebinding and cannot modify system namespace like (kube-system, kyma-system, istio-system)?

Comment (Demonsthere) [reply tests/cluster-users-test/sar-test.sh]: Implemented, please verify

**Pull request merged**